### PR TITLE
Use pure rust libraries for regex and snakecasing

### PR DESCRIPTION
**Pull request by acmiyaguchi**
This PR uses regex and [heck](https://docs.rs/heck/0.3.3/heck/) as replacements for [oniguruma](https://github.com/rust-onig/rust-onig). Oniguruma is used to perform the snake casing logic in a language independent way. The rules were originally sourced from heck, and are fully compatible with the casing tests.

This should reduce build times (since we no longer have to compile oniguruma from the C sources) and make it easier to compile the package to wasm.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,12 +1,8 @@
 #![recursion_limit = "128"]
 #[macro_use]
 extern crate log;
-extern crate onig;
 #[macro_use]
 extern crate serde;
-extern crate serde_json;
```
Comment (relud) [inline src/lib.rs:7]: is this line intentionally removed? it doesn't look related to any of the other changes

**Review by relud — APPROVED** [review]
r+wc

File: src/lib.rs (hunk 1 above)
Comment (acmiyaguchi) [reply src/lib.rs:7]: I ended up removing this since the extern crate declaration isn't necessary anymore with the 2018 edition, and I was moving the `lazy_static`/`onig` declarations.
